packages/db/src/meta/pouch/types.ts

```diff
@@ -6,6 +6,81 @@ import PouchDB from "pouchdb";
 import type { Collections, CollectionName } from "@truffle/db/meta/collections";
 import type * as Id from "@truffle/db/meta/id";
 
+type Identified = PouchDB.Core.IdMeta;
+type Retrieved = PouchDB.Core.GetMeta;
+
+/**
+ * An object is a Record<T> if it has identification info
+ * (i.e., `{ _id: string }`)
+ */
+export type Record<T> = T & Identified;
+
+/**
+ * SavedRecord<T> objects additionally contain PouchDB revision info, etc.
+ */
+export type SavedRecord<T> = Record<T> & Retrieved;
+
+/**
+ * A reference has only identification information
+ */
+export type RecordReference<T> = Pick<Record<T>, "_id">;
+
+/**
+ * Low-level interface for PouchDB adapters - allows basic CRUD operations
+ * for arbitrary (weakly-typed) data types.
+ */
+export interface Adapter<C extends Collections> {
+  /**
+   * Retrieve and return every saved record for a given collection name
+   */
+  every<N extends CollectionName<C>, T>(
```

Comment:
What are these function names based on? Is there guidance somewhere on how these CRUD-y functions should be named that you based this on? I know you can't use the `all` because you need it in the workspace file, just wondering how you arrived at this naming given that limitation.

Reply:
Gosh Faina I was just trying to be cute with naming cause I needed a bunch of verbs. We can use standard ones I suppose :)

Probably also doesn't matter much if some of the verbs collide with some of the verbs that Workspace uses? I had implemented them all in the same file to start, so I couldn't have collisions at that point and never changed it.